### DISCUSSION TITLE
Online TV from Facebook pages

### DISCUSSION THREAD

**Discussion by sguinetti** [iptv-org/community#273] [Show and tell]
In Google, searching " "tv en vivo" site:facebook.com/*/app " finds some Facebook page with online tv (insead of official website page because they don't have). Examples from Peru (with "Peru" in search) are:

* https://www.facebook.com/radiotvtocache/app/349313058487732/
* https://www.facebook.com/SicsibambaSitioOficial/app/14167664298
* https://www.facebook.com/RadioMixCajatambo/app/212104595551052/
* https://www.facebook.com/forotvpe/app/189116767802011
* https://www.facebook.com/solartvcusc/app/208195102528120
* https://www.facebook.com/manantialdevidape/app/190322544333196
* https://www.facebook.com/cncsenaldigital/app/208743069846461

Other prolifes are reference with stream links added in pe.m3u:
* https://www.facebook.com/radiosanborja/app/190322544333196 
* https://www.facebook.com/geniostvmoyobamba/app/190322544333196
* https://www.facebook.com/BHtvNuestroCanal/app/190322544333196/
* https://www.facebook.com/FUEGOTVLIMA/app/337680106275807

Can you test if this link have some m3u link? Facebook is problematic to show these links.

**Comment by BellezaEmporium** [comment]
Those are DASH streams, they might be taken using Streamlink if you give them a persistent "live stream" page.

Streamlink's code indicate this : 
```
@pluginmatcher(re.compile(r"""
    https?://(?:www\.)?facebook
    (?:\.com|wkhpilnemxj7asaniu7vnjjbiltxjqhye3mhbshg7kx5tfyd\.onion)
    /[^/]+/(?:posts|videos)/(?P<video_id>\d+)
""", re.VERBOSE))
```

Not having any Facebook account myself, i'll have to see using Streamlink if I can gather stuff.

**Comment by BellezaEmporium** [comment]
> 
>     * https://www.facebook.com/radiotvtocache/app/349313058487732/ = https://sonicpanel.tmcreativos.com/8044/stream only radio available
> 
>     * https://www.facebook.com/SicsibambaSitioOficial/app/14167664298 = https://apps1.af-design.com/tabs_eol 404
> 
>     * https://www.facebook.com/RadioMixCajatambo/app/212104595551052/ = Redirects to a youtube video
> 
>     * https://www.facebook.com/forotvpe/app/189116767802011 = http://free.fullspeed.tv/iptv-query?streaming-ip=https://www.ustream.tv/embed/18196244?html5ui
> 
>     * https://www.facebook.com/solartvcusc/app/208195102528120 = https://srv4.zcast.com.br/sistemas2020/sistemas2020/playlist.m3u8
> 
>     * https://www.facebook.com/manantialdevidape/app/190322544333196 = https://raw2.statichtmlapp.com/tab/1/visitor/266889696761374 POST on 2 different parts of the server, stream doesn't work;
> 
>     * https://www.facebook.com/cncsenaldigital/app/208743069846461 = http://167.114.116.212:1935/adestv/myStream/playlist.m3u8 / rtsp://167.114.116.212:1935/adestv/myStream
>
>     * https://www.facebook.com/radiosanborja/app/190322544333196 = https://5c3fb01839654.streamlock.net:1963/iptvsanborja/livesanborjatv/playlist.m3u8
> 
>     * https://www.facebook.com/geniostvmoyobamba/app/190322544333196 = https://5b4920f622b50.streamlock.net/geniostv/geniostv/playlist.m3u8
> 
>     * https://www.facebook.com/BHtvNuestroCanal/app/190322544333196/ = https://5c3fb01839654.streamlock.net:1963/iptvbhtv/livebhtvtv/playlist.m3u8
> 
>     * https://www.facebook.com/FUEGOTVLIMA/app/337680106275807 (https://www.facebook.com/Fuegotv.pe/app/337680106275807) = https://stmv1.zcastbr.com/fuegotvperu/fuegotvperu/playlist.m3u8
>

**Reply by sguinetti** [reply]
Thanks for find these links (And happy new year).

**Comment by BellezaEmporium** [comment]
Updated FB playlists.
